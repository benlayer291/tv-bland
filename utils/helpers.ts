export const removeProtocolFromUrl = (url: string): string =>
  url.replace(/^https?:\/\//i, '//')

export const removeHtmlTagsFromString = (htmlString: string): string =>
  htmlString.replace(/(<([^>]+)>)/gi, '')

export const checkDeviceIsMobile = (): boolean => {
  return (
    typeof navigator !== 'undefined' &&
    /Android|Mobi/i.test(navigator.userAgent)
  )
}
