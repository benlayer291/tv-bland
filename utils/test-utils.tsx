import { render, RenderOptions } from '@testing-library/react'

type Props = {
  children: React.ReactElement
}

const Providers: React.FunctionComponent<Props> = ({ children }: Props) => {
  return <>{children}</>
}

const customRender = (
  ui: React.ReactElement,
  options: Omit<RenderOptions, 'queries'> = {}
) => render(ui, { wrapper: Providers, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }
