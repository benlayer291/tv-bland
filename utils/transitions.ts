type Transtion = {
  duration: number
  ease: number[]
}

export const transition: Transtion = {
  duration: 5,
  ease: [0.43, 0.13, 0.23, 0.96],
}
