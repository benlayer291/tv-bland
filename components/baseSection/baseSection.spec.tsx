import React from 'react'
import { render, RenderResult } from '../../utils/test-utils'

import BaseSection from './baseSection'

const children = 'Children'

const init = (props?: Record<string, unknown>): RenderResult => {
  const propsInit = { ...props, children }

  return render(<BaseSection {...propsInit} />)
}

describe('<BaseSection />', () => {
  let wrapper: RenderResult

  beforeEach(() => (wrapper = init()))

  it('renders correctly', () => {
    expect(wrapper.container).toBeInTheDocument()
  })

  it('renders children', () => {
    expect(wrapper.container).toContainHTML(children)
  })

  it('matches snapshot', () => {
    expect(wrapper.container).toMatchSnapshot()
  })
})
