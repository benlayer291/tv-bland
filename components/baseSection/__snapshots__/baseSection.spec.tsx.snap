// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BaseButton /> matches snapshot 1`] = `
<div>
  <section>
    <div
      class="wrapper"
    >
      Children
    </div>
  </section>
</div>
`;
